File: integration-tests/mongodb-panache/src/main/java/io/quarkus/it/mongodb/panache/bugs/LinkedEntity.java
package io.quarkus.it.mongodb.panache.bugs;

import org.bson.types.ObjectId;

import io.quarkus.mongodb.panache.PanacheMongoEntity;

public class LinkedEntity extends PanacheMongoEntity {
    public String name;
    public ObjectId myForeignId;
}

File: integration-tests/mongodb-panache-kotlin/src/main/kotlin/io/quarkus/it/mongodb/panache/bugs/LinkedEntity.kt
package io.quarkus.it.mongodb.panache.bugs

import io.quarkus.mongodb.panache.kotlin.PanacheMongoCompanion
import io.quarkus.mongodb.panache.kotlin.PanacheMongoEntity
import org.bson.types.ObjectId

class LinkedEntity : PanacheMongoEntity() {
    companion object: PanacheMongoCompanion<LinkedEntity>

    var name: String? = null
    var myForeignId: ObjectId? = null
}